File: android/StockAnalyzer/app/src/main/java/com/stockanalyzer/data/model/Stock.kt
package com.stockanalyzer.data.model

import com.google.gson.annotations.SerializedName

data class StockSearchResult(
    val code: String,
    val name: String,
    val market: String,
    val price: Double?,
    @SerializedName("change_pct") val changePct: Double?
)

data class StockQuote(
    val code: String,
    val name: String,
    val market: String,
    val price: Double,
    val change: Double,
    @SerializedName("change_pct") val changePct: Double,
    val volume: Double,
    val amount: Double,
    val high: Double,
    val low: Double,
    val open: Double,
    @SerializedName("prev_close") val prevClose: Double
)

data class KLineBar(
    val date: String,
    val open: Double,
    val high: Double,
    val low: Double,
    val close: Double,
    val volume: Double
)

data class TechnicalIndicators(
    val ma5: Double?, val ma10: Double?, val ma20: Double?, val ma60: Double?,
    @SerializedName("macd_dif") val macdDif: Double?,
    @SerializedName("macd_dea") val macdDea: Double?,
    @SerializedName("macd_bar") val macdBar: Double?,
    val rsi6: Double?, val rsi12: Double?, val rsi24: Double?,
    @SerializedName("kdj_k") val kdjK: Double?,
    @SerializedName("kdj_d") val kdjD: Double?,
    @SerializedName("kdj_j") val kdjJ: Double?,
    @SerializedName("boll_upper") val bollUpper: Double?,
    @SerializedName("boll_mid") val bollMid: Double?,
    @SerializedName("boll_lower") val bollLower: Double?,
    @SerializedName("volume_ratio") val volumeRatio: Double?,
    @SerializedName("turnover_rate") val turnoverRate: Double?,
    val score: Int
)

data class FundamentalData(
    @SerializedName("pe_ttm") val peTtm: Double?,
    val pb: Double?, val ps: Double?,
    val roe: Double?, val roa: Double?,
    @SerializedName("revenue_growth_yoy") val revenueGrowthYoy: Double?,
    @SerializedName("profit_growth_yoy") val profitGrowthYoy: Double?,
    @SerializedName("gross_margin") val grossMargin: Double?,
    @SerializedName("net_margin") val netMargin: Double?,
    @SerializedName("debt_ratio") val debtRatio: Double?,
    @SerializedName("current_ratio") val currentRatio: Double?,
    @SerializedName("free_cash_flow") val freeCashFlow: Double?,
    val eps: Double?,
    @SerializedName("dividend_yield") val dividendYield: Double?,
    val score: Int
)

data class ValuationData(
    @SerializedName("industry_avg_pe") val industryAvgPe: Double?,
    @SerializedName("industry_avg_pb") val industryAvgPb: Double?,
    @SerializedName("pe_percentile") val pePercentile: Double?,
    @SerializedName("pb_percentile") val pbPercentile: Double?,
    val peg: Double?,
    val score: Int
)

data class SentimentData(
    @SerializedName("main_net_inflow") val mainNetInflow: Double?,
    @SerializedName("super_large_net") val superLargeNet: Double?,
    @SerializedName("large_net") val largeNet: Double?,
    @SerializedName("northbound_net") val northboundNet: Double?,
    @SerializedName("margin_balance") val marginBalance: Double?,
    @SerializedName("news_heat_score") val newsHeatScore: Int,
    @SerializedName("is_limit_up") val isLimitUp: Boolean,
    @SerializedName("is_limit_down") val isLimitDown: Boolean,
    val score: Int
)

data class ChipData(
    @SerializedName("institution_holding_pct") val institutionHoldingPct: Double?,
    @SerializedName("shareholder_count_change_pct") val shareholderCountChangePct: Double?,
    @SerializedName("top10_holder_change") val top10HolderChange: String?,
    val score: Int
)

data class MacroData(
    @SerializedName("industry_name") val industryName: String?,
    @SerializedName("industry_change_pct") val industryChangePct: Double?,
    @SerializedName("corr_with_index") val corrWithIndex: Double?,
    @SerializedName("policy_summary") val policySummary: String?,
    val score: Int
)

data class StockAnalysis(
    val code: String, val name: String, val market: String,
    val technical: TechnicalIndicators,
    val fundamental: FundamentalData,
    val valuation: ValuationData,
    val sentiment: SentimentData,
    val chip: ChipData,
    val macro: MacroData
)

data class AIReport(
    val code: String, val name: String, val market: String,
    val summary: String,
    @SerializedName("market_review") val marketReview: String,
    @SerializedName("fundamental_analysis") val fundamentalAnalysis: String,
    @SerializedName("technical_analysis") val technicalAnalysis: String,
    @SerializedName("valuation_analysis") val valuationAnalysis: String,
    @SerializedName("sentiment_analysis") val sentimentAnalysis: String,
    val risks: List<String>,
    val recommendation: String,
    val rating: String,
    @SerializedName("target_price_low") val targetPriceLow: Double?,
    @SerializedName("target_price_high") val targetPriceHigh: Double?,
    @SerializedName("generated_at") val generatedAt: String,
    val cached: Boolean
)

File: models/stock.py
from pydantic import BaseModel
from typing import Optional

class StockSearchResult(BaseModel):
    code: str
    name: str
    market: str  # A | HK | US
    price: Optional[float] = None
    change_pct: Optional[float] = None

class StockQuote(BaseModel):
    code: str
    name: str
    market: str
    price: float
    change: float
    change_pct: float
    volume: float
    amount: float
    high: float
    low: float
    open: float
    prev_close: float

class KLineBar(BaseModel):
    date: str
    open: float
    high: float
    low: float
    close: float
    volume: float

class TechnicalIndicators(BaseModel):
    ma5: Optional[float] = None
    ma10: Optional[float] = None
    ma20: Optional[float] = None
    ma60: Optional[float] = None
    macd_dif: Optional[float] = None
    macd_dea: Optional[float] = None
    macd_bar: Optional[float] = None
    rsi6: Optional[float] = None
    rsi12: Optional[float] = None
    rsi24: Optional[float] = None
    kdj_k: Optional[float] = None
    kdj_d: Optional[float] = None
    kdj_j: Optional[float] = None
    boll_upper: Optional[float] = None
    boll_mid: Optional[float] = None
    boll_lower: Optional[float] = None
    volume_ratio: Optional[float] = None
    turnover_rate: Optional[float] = None
    score: int = 50

class FundamentalData(BaseModel):
    pe_ttm: Optional[float] = None
    pb: Optional[float] = None
    ps: Optional[float] = None
    roe: Optional[float] = None
    roa: Optional[float] = None
    revenue_growth_yoy: Optional[float] = None
    profit_growth_yoy: Optional[float] = None
    gross_margin: Optional[float] = None
    net_margin: Optional[float] = None
    debt_ratio: Optional[float] = None
    current_ratio: Optional[float] = None
    free_cash_flow: Optional[float] = None
    eps: Optional[float] = None
    dividend_yield: Optional[float] = None
    score: int = 50

class ValuationData(BaseModel):
    industry_avg_pe: Optional[float] = None
    industry_avg_pb: Optional[float] = None
    pe_percentile: Optional[float] = None
    pb_percentile: Optional[float] = None
    peg: Optional[float] = None
    score: int = 50

class SentimentData(BaseModel):
    main_net_inflow: Optional[float] = None
    super_large_net: Optional[float] = None
    large_net: Optional[float] = None
    northbound_net: Optional[float] = None
    margin_balance: Optional[float] = None
    news_heat_score: int = 50
    is_limit_up: bool = False
    is_limit_down: bool = False
    score: int = 50

class ChipData(BaseModel):
    institution_holding_pct: Optional[float] = None
    shareholder_count_change_pct: Optional[float] = None
    top10_holder_change: Optional[str] = None
    score: int = 50

class MacroData(BaseModel):
    industry_name: Optional[str] = None
    industry_change_pct: Optional[float] = None
    corr_with_index: Optional[float] = None
    policy_summary: Optional[str] = None
    score: int = 50

class StockAnalysis(BaseModel):
    code: str
    name: str
    market: str
    technical: TechnicalIndicators
    fundamental: FundamentalData
    valuation: ValuationData
    sentiment: SentimentData
    chip: ChipData
    macro: MacroData

class AIReport(BaseModel):
    code: str
    name: str
    market: str
    summary: str
    market_review: str
    fundamental_analysis: str
    technical_analysis: str
    valuation_analysis: str
    sentiment_analysis: str
    risks: list[str]
    recommendation: str
    rating: str
    target_price_low: Optional[float] = None
    target_price_high: Optional[float] = None
    generated_at: str
    cached: bool = False
